### ISSUE TITLE
Resolver: the resolver does not parse Kattis's penalty_time field

### ISSUE THREAD

**Issue by evouga** [icpctools/icpctools#1074]
Kattis emits `penalty_time` in the new CCS format, which is `(-)?(h)*h:mm:ss(.uuu)?` rather than an integer.

The Resolver does not parse this format, and uses 0 as the penalty time instead. This caused the Resolver to show incorrect team rankings this year at the South Central Regional.

Also including @niemela

**Comment by deboer-tim** [comment]
FWIW, new penalty time is in 'future draft' state in the spec: https://ccs-specs.icpc.io/draft/contest_api#contests. The tools will support it soon when we close the spec and have a contest to test at, but I'm sorry if you weren't aware that's not now.

**Comment by niemela** [comment]
It would be really nice if there could be a warning when running against and API and the version provided (as specified in version property of the API information endpoint) is not what was expected. 

Simply something like "Expected version 2023-06, got version draft" would be really useful.

**Comment by evouga** [comment]
Yes, I agree! In fact I would suggest an error rather than a warning, because there are already a lot of benign warnings when you run the Resolver, and at first glance the current Resolver seems to work fine on the Kattis output (you have to notice that the time penalties are wrong compared to the scoreboard).

> The tools will support it soon when we close the spec and have a contest to test at, but I'm sorry if you weren't aware that's not now.

No worries---I understand it's not reasonable to expect the ICPC Tools to accept inputs that do not adhere to the current spec. That said, as a contest director it would be nice if the ICPC Tools and CCSs interoperate in a way that Just Works to the extent possible, and fail in an obvious way when not.

**Comment by deboer-tim** [comment]
A few updates:
- All the tools started supporting what became the 2026-01 spec a long time ago (sorry I forgot to ping here), and most of the current 2026-draft. So this specific issue wouldn't have happened for over a year.
- As of #1326 yesterday, the tools check the API version and warn if it's not a supported version.
- 'draft' in 2024 meant 'anything that wasn't a release', and the CDS did support that (it supported *lots* of draft versions of the spec, just not the latest one Kattis did), so you still wouldn't have received a warning. Spec clarification on drafts will fix this as much as possible going forward: https://github.com/icpc/ccs-specs/issues/302

So to sum up, once the spec issue is complete I can change the warning to error, and everything should finally be as complete as it can be here.
